app/models/project.rb

```diff
@@ -185,8 +185,13 @@ def self.send_verify_moip_account_notification
     end
   end
 
-  def self.goal_between(starts_at, ends_at)
-    where("goal BETWEEN ? AND ?", starts_at, ends_at)
+  def self.goal_between(start_at=nil, ends_at=nil)
+    max_integer = (2**(0.size * 8 -2) -1)
+
+    start_at ||= 0
+    ends_at  ||= max_integer
+
+    where(goal: start_at.to_i..ends_at.to_i)
```

Comment:
Why don't you just:

```ruby
scope :goal_between, -> (value_starts_at = nil, values_ends_at = nil) {
  where(goal: value_starts_at, value_ends_at) if value_starts_at || value_ends_at
}
```

You don't need to return an Active Record relation from within a scope, scopes automatically manage to return relations which you can use to chain with other scopes (even if you return nil).

Correct me if I'm wrong, but I guess your intention with these "high" default numbers is to avoid doing conditionals and to provide a number high enough to match all projects. A small conditional with a scope does not hurt and ends up having the same effect.

Can you try this out and see if your spec still passes?